### PR TITLE
1596 - MVP for genAI service support

### PR DESCRIPTION
**Pull request by Vladysl**

### CODE REVIEW

File: odd-platform-specification/components.yaml [hunk 1]
```diff
@@ -4063,6 +4063,18 @@ components:
...
+      type: object
+      properties:
+        question:
+          type: string
+
+    GenAIQuestionResult:
+      type: object
+      properties:
+        question_response:
+          type: string
```
Comment (AndreyNenashev) [inline odd-platform-specification/components.yaml:4076]: components and properties naming feels a little bit redundant
```suggestion
    GenAIRequest:
      type: object
      properties:
        body:
          type: string

    GenAIResponse:
      type: object
      properties:
        body:
          type: string
```

File: odd-platform-specification/openapi.yaml [hunk 2]
```diff
@@ -3883,4 +3884,25 @@ paths:
               schema:
                 $ref: './components.yaml/#/components/schemas/DataEntityGraphRelationshipDetails'
       tags:
-        - relationship
\ No newline at end of file
+        - relationship
+
+  /api/genai/question:
```
Comment (AndreyNenashev) [inline odd-platform-specification/openapi.yaml:3889]: you can ignore it
**nit**: question is quite strict word. How about `ask` there is no guarantee that it's gonna be just questioning. It's also requests like "give me example of `something` configuration example"
```suggestion
  /api/genai/ask:
```

**Review by AndreyNenashev — CHANGES_REQUESTED** [review]

File: odd-platform-specification/components.yaml (hunk 1 above)
Comment (Vladysl) [reply odd-platform-specification/components.yaml]: done

File: odd-platform-specification/openapi.yaml (hunk 2 above)
Comment (Vladysl) [reply odd-platform-specification/openapi.yaml]: done

**Review by AndreyNenashev — APPROVED** [review]

File: odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/service/genai/GenAIServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+@Slf4j
+@Service
+public class GenAIServiceImpl implements GenAIService {
+    public static final String QUERY_DATA = "/query_data";
+    public static final String QUESTION_FIELD = "question";
```
Comment (AndreyNenashev) [inline odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/service/genai/GenAIServiceImpl.java:24]: I think it must be "body" now according the spec

**Review by AndreyNenashev — CHANGES_REQUESTED** [review]

**Review by AndreyNenashev — APPROVED** [review]

**Pull request merged**